Pull request to merge emergency-weather-monitor to develop branch

jermainegug opened the pull request:
Only issues I am having with this code is that for the 2.3m and SkyMapper, the rain condition is always showing as invalid. I have checked the files and code but I can't see why it would assign it to invalid.

jermainegug committed:
Add metdata weather & local config file

jermainegug committed:
Adjust comments and documentation

jermainegug committed:
Start to form base weather class

* minor comment change in weather_ext.py
* delete the check past 15 minute for rain in weather_ext.py
* create and make common function in weather_abstract.py

jermainegug committed:
Write some code in abstract weather

* edit the weather files to work with abstact weather
* add stuff to config file to make more generic

jermainegug committed:
Import ABC

jermainegug committed:
Multiple products in config

* import `get_mongodb` in weather files
* put libraries back into `weather_ext.py`

jermainegug committed:
Edit weather files

jermainegug committed:
Edit capture method

jermainegug committed:
Allow for data to be included in safety decision

* for the safety condition methods the current_values parameter was included so that the metdata does not have to be fetched each time capture calls on them

NOTE: If the base class doesn't work use the code at this stage to use with the dome.

jermainegug committed:
Merge branch 'improve-weather-code' into emergency-weather-monitor

jermainegug committed:
post-merge cleanup

* better base methods
* more generic
* should work

jermainegug committed:
Check to make sure num col = num units specified

jermainegug committed:
Merge branch 'improve-weather-code' into origin/improve-weather-code

jermainegug committed:
pre-pre-merge cleanup (weather-monitor to develop)

jermainegug committed:
Fix rain safety classes

* majority of the code is in the base class
* code in `weather.py` truns the AAG sensor's rain values into binary values to work with the base class

jermainegug committed:
Add arguments to rain safety base class

jermainegug committed:
Edit thresholds in config file and make necessary changes

* rain safety check was edited to make all entries ~binary for rain condition (needs to be changed anyway)
* `simple_weather_capture.py` incldes code to be able to test on windows laptop (needs to be changed)

jermainegug committed:
Replace previous rain safety method with a draft

* adjust comments in `simple_weather_capture.py`

jermainegug committed:
Add rain safety code to get condition and safety 

* adjusted config file to allow for infinity values to work cut out the clear/calm conditions which were not used in the code.
* fix weather and weather_web_data files, correct super().'s, add rain safety checks.
* fix weather abstract file, correct methods
*import weather_web_data to the simple_weather_capture file to run both the files for aag and aat data.

jermainegug committed:
Add 'clear', 'calm', etc. to thresholds in config

jermainegug committed:
Make changes to code in `weather_abstract.py`

* edit capture method so that it is more understandable
* edit make_safety_decision method so that it is more concise and generic
* _get_safety methods changed to be more concise and include the invalid along with an unknown condition
* edit _get_status method back to invalid from unknown

jermainegug committed:
Fix typo's and adjust some methods

jermainegug committed:
Change file name and minor edits

* delete _get_cloud_safety and _get_wind_safety methods - they become obsolete as the methods in base class covered all aspects of derived methods

jermainegug committed:
Edit config and make changes to wind, gust and cloud methods

* not sure if the way I have implemented self.weather_entries['...'] = value, is correct. Doesn't seem right but not sure what else to do.

jermainegug committed:
Fix typo and file name change

* updated latptop: path was changed

jermainegug committed:
Correcting code in derived weather classes

jermainegug committed:
Added base method for rain safety

Note that this method is not implemented due to the AAG and AAT having different thresholds for their rain data.

jermainegug committed:
Change confusing data variable to current_values for capture method

This is for both dervied methods

jermainegug committed:
Fix typo's

jermainegug committed:
Fixing up code

jermainegug committed:
Fixed errors that came up when running

* When I run the code it doesn't give me anything back - It glitches and I have to shut down the prompt window...

jermainegug committed:
Big clean up

* added comments - may need to be reviewed
* fixed names
* fixed up simple_weather_capture.py

jermainegug committed:
Typo fixes and adjust simple weather capture

jermainegug committed:
Allow for simple weather capture to write all weather data to file

jermainegug committed:
Fixes

jermainegug committed:
Edit so that it actually works

jermainegug committed:
Working files!

jermainegug committed:
Make names more readable

*add back write header (not currently used)
*change file name for aat metdata class

jermainegug committed:
Changing from web_service to aat_metdata

If we want to eventually add more weather data, perhaps from the 2.3 m and the sky-mapper.

jermainegug committed:
Small edit and change

jermainegug committed:
Now we know what the 120. is

jermainegug committed:
Fix typo in config

jermainegug committed:
Fixes

jermainegug committed:
Start to make files and add stuff to weather capture

jermainegug committed:
Fix to open file

jermainegug committed:
Name and serial port change

jermainegug committed:
Edits and some adjustment

jermainegug committed:
Change to test AAG temporarily

jermainegug committed:
Added more code to get weather data

*added names, thresholds to config for skymap and 2.3m
*added function to print the data
*editied files to successfuly get important info from skymap and 2.3m

ISSUE: Rain data is always coming up invalid in the code and I can't figure out why :(

jermainegug committed:
Fix datetime formatting issue

*fixed the time issue and now only downloads the data when it has to :)

jermainegug committed:
add nice new line to separate data in file

jermainegug committed:
Merge branch 'emergency-weather-monitor' into get_met23_and_skymap

jermainegug committed:
Fixed weird lines

Obviously the lines ended up being added when I tried to resolve the file conflicts but I have fixed the lines to what we want.

jermainegug committed:
Merge pull request #4 from AstroHuntsman/get_met23_and_skymap

Merge get_met23_and_skymap to emergency-weather-monitor

jermainegug committed:
Comments for 2.3m and SkyMap files

jermainegug committed:
Small adjustments

jermainegug committed:
Bring back code to check AAG

*make note of `import sys`

jermainegug committed:
Fix weather data from 2.3m and SkyMapper

*have a working code and should be good to use.
*still needs some tidying up in some places.

jermainegug committed:
Update files to parse 2.3m and SkyMap data into a table

*added all entries of the 2.3m and SkyMap to config and classes
*name changes
*

jermainegug committed:
Fixed up rain sensor issue for skymap

jermainegug committed:
Make things look good and give own times

jermainegug committed:
Get rid of annoying new line header

jermainegug committed:
Small adjust

*got rid of useless line

jermainegug commented:
Previous issues with the coed have been fixed. Should be ready to merge!

wtgee commented on config.yaml:
```diff
@@ -13,8 +13,174 @@ environment:
 weather:
     station: mongo
+    aat_metdata:
+      name: 'AAT metdata'
```
FYI, you don't actually need the quotes in YAML and then it looks a little cleaner.

jermainegug replied:
Okay, is that the same for the quotes that are in the array?

wtgee replied:
Yes. You only need quotes if you need to force a string, like

```
version: "10.1"
```

wtgee commented on config.yaml:
```diff
@@ -13,8 +13,174 @@ environment:
+                      'boltwood_wet_flag', # dry = 0 | wet > 0
+                      'rain_since_9am',
+                      'SQM_sky_brightness']
+      # be sure to have the same number of entries in column_names and column_units
```
If column names and units always have to match you can make them a dictionary instead of two lists:

```
columns:
    time_UTC: u.dimensionless_unscaled,
    outside_air_temparature: u.Celsius
    ...etc
```

You will then have to read it in slightly different, getting the `.keys()` or `.values()` (or `.items()`) depending on how you use it.
 
This would be preferred as it would enforce matching entries and make it a little more obvious what goes with what.

wtgee reviewed (commented):
Started review but I'm going to pull down code and try to use it then will finish review. Submitting two comments now.

jermainegug committed:
Better assignment of units to items

*changed config file to be more readable
*adjust methods to work with new config layout

jermainegug committed:
Fix typo
